File: SampleApp/java/app/src/main/java/jp/fluct/sample/samplefluctsdkapp/VideoInterstitialActivity.java
package jp.fluct.sample.samplefluctsdkapp;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Calendar;
import java.util.GregorianCalendar;

import jp.fluct.fluctsdk.FluctAdRequestTargeting;
import jp.fluct.fluctsdk.FluctErrorCode;
import jp.fluct.fluctsdk.FluctVideoInterstitial;
import jp.fluct.fluctsdk.FluctVideoInterstitialSettings;

public class VideoInterstitialActivity extends AppCompatActivity {
    private FluctVideoInterstitial rewardedVideo;
    private Button showButton;
    private TextView stateTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_interstitial);

        stateTextView = (TextView) findViewById(R.id.state_textview);

        String groupID = "1000120112";
        String unitID = "1000207345";

        FluctVideoInterstitialSettings settings = new FluctVideoInterstitialSettings.Builder()
                .testMode(true)
                .debugMode(true)
                .build();

        rewardedVideo = FluctVideoInterstitial.getInstance(groupID, unitID, this, settings);

        rewardedVideo.setListener(new FluctVideoInterstitial.Listener() {
            // 広告読み込み完了
            @Override
            public void onLoaded(String groupId, String unitId) {
                updateStateText("Loaded");
                showButton.setEnabled(true);
                Log.d("RewardedVideoActivity", "onLoaded");
            }

            // 広告読み込み失敗
            @Override
            public void onFailedToLoad(String groupId, String unitId, FluctErrorCode errorCode) {
                updateStateText("Failed to load");
                Log.d("RewardedVideoActivity", "onFailedToLoad");
            }

            // 広告が表示した
            @Override
            public void onOpened(String groupId, String unitId) {
                Log.d("RewardedVideoActivity", "onOpened");
            }

            // 動画が再生された
            @Override
            public void onStarted(String groupId, String unitId) {
                Log.d("RewardedVideoActivity", "onStarted");
            }

            // 広告が閉じられた
            @Override
            public void onClosed(String groupId, String unitId) {
                updateStateText("Complete");
                showButton.setEnabled(false);
                Log.d("RewardedVideoActivity", "onClosed");
            }

            // 広告の再生が失敗した
            @Override
            public void onFailedToPlay(String groupId, String unitId, FluctErrorCode errorCode) {
                updateStateText("Failed to play");
                Log.d("RewardedVideoActivity", "onFailedToPlay");
            }
        });

        showButton = (Button) findViewById(R.id.show_button);
        showButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!rewardedVideo.isAdLoaded()) {
                    return;
                }

                rewardedVideo.show();
            }
        });

        findViewById(R.id.load_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FluctAdRequestTargeting targeting = new FluctAdRequestTargeting();
                targeting.setGender(FluctAdRequestTargeting.FluctGender.MALE);
                targeting.setBirthday(new GregorianCalendar(1988, Calendar.JANUARY, 1).getTime());
                rewardedVideo.loadAd(targeting);
                updateStateText("Loading");
            }
        });
    }

    @SuppressLint("SetTextI18n")
    private void updateStateText(String s) {
        stateTextView.setText("State: " + s);
    }

}

File: SampleApp/kotlin/app/src/main/java/jp/fluct/sample/kotlinapp/VideoInterstitialActivity.kt
package jp.fluct.sample.kotlinapp

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Button
import android.widget.TextView
import jp.fluct.fluctsdk.FluctAdRequestTargeting
import jp.fluct.fluctsdk.FluctErrorCode
import jp.fluct.fluctsdk.FluctVideoInterstitial
import jp.fluct.fluctsdk.FluctVideoInterstitialSettings
import java.util.*

class VideoInterstitialActivity : AppCompatActivity() {

    private lateinit var rewardedVideo: FluctVideoInterstitial

    private var stateTextView: TextView? = null
    private var showButton: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rewardedvideo)

        stateTextView = findViewById<TextView>(R.id.state_textview)

        val groupID = "1000120112"
        val unitID = "1000207345"

        val settings = FluctVideoInterstitialSettings.Builder()
                .testMode(true)
                .debugMode(true)
                .build()

        rewardedVideo = FluctVideoInterstitial.getInstance(groupID, unitID, this, settings)
        rewardedVideo.setListener(object : FluctVideoInterstitial.Listener {
            // 広告読み込み完了
            override fun onLoaded(groupId: String?, unitId: String?) {
                showButton?.isEnabled = true
                updateStateTextView("Loaded")
                Log.d("RewardedVideoActivity", "onLoaded")
            }

            // 広告読み込み失敗
            override fun onFailedToLoad(groupId: String?, unitId: String?, errorCode: FluctErrorCode?) {
                updateStateTextView("Failed to load")
                Log.d("RewardedVideoActivity", "onFailedToLoad")
            }

            // 広告が表示した
            override fun onOpened(groupId: String?, unitId: String?) {
                Log.d("RewardedVideoActivity", "onOpened")
            }

            // 動画が再生された
            override fun onStarted(groupId: String?, unitId: String?) {
                Log.d("RewardedVideoActivity", "onStarted")
            }

            // 広告が閉じられた
            override fun onClosed(groupId: String?, unitId: String?) {
                showButton?.isEnabled = false
                updateStateTextView("Complete")
                Log.d("RewardedVideoActivity", "onClosed")
            }

            // 広告の再生が失敗した
            override fun onFailedToPlay(groupId: String?, unitId: String?, errorCode: FluctErrorCode?) {
                Log.d("RewardedVideoActivity", "onFailedToPlay")
            }
        })

        showButton = findViewById<Button>(R.id.show_button)
        showButton?.setOnClickListener {
            if (rewardedVideo.isAdLoaded) {
                rewardedVideo.show()
            }
        }
        findViewById<Button>(R.id.load_button).setOnClickListener {
            val targeting = FluctAdRequestTargeting()
            val userID = "APP_USER_ID"
            targeting.setUserId(userID)
            targeting.gender = FluctAdRequestTargeting.FluctGender.MALE
            targeting.birthday = GregorianCalendar(1988, Calendar.JANUARY, 1).time
            rewardedVideo.loadAd(targeting)
            updateStateTextView("Loading")
        }
    }

    @SuppressLint("SetTextI18n")
    fun updateStateTextView(text: String) {
        stateTextView?.setText("State: " + text)
    }
}
